app/battle/pre_battle.py

```diff
@@ -0,0 +1,17 @@
+def pre_battle(knights_one: dict) -> dict:

```

Comment:
It would be better to use more descriptive naming for this function and file, e.g. `prepare_knights`.